editor/plugins/node_3d_editor_plugin.h

```diff
@@ -715,7 +715,7 @@ class Node3DEditor : public VBoxContainer {
 
 	Node3D *selected = nullptr;
 
-	void _request_gizmo(Object *p_obj);
+	void _request_gizmo(ObjectID p_object);
```

Comment:
Instead of changing the signature, why not accept a `Object`, but calling `Object::get_instance_id()` inside `_request_gizmo()`?

Reply:
You are right. The following also works.
``` diff
diff --git a/editor/plugins/node_3d_editor_plugin.cpp b/editor/plugins/node_3d_editor_plugin.cpp
index 44126b4497..dd2d925612 100644
--- a/editor/plugins/node_3d_editor_plugin.cpp
+++ b/editor/plugins/node_3d_editor_plugin.cpp
@@ -7599,6 +7599,9 @@ void Node3DEditor::move_control_to_right_panel(Control *p_control) {
 }
 
 void Node3DEditor::_request_gizmo(Object *p_obj) {
+       if (p_obj->get_instance_id().is_null()) {
+               return;
+       }
        Node3D *sp = Object::cast_to<Node3D>(p_obj);
        if (!sp) {
                return;
```

Reply:
Wait, `is_null()` does only return true if its `id` is equal to `0`

Reply:
Because the passed object may be invalid. That's the whole point of using ObjectID.
Undo stores method + arguments and the object in argument can be freed at some point.

Reply:
> Wait, `is_null()` does only return true if its `id` is equal to `0`. I think you should keep `ObjectDB::get_instance(p_object)` and compare it to `nullptr`.

This case is covered in my PR, because in the following line `sp = nullptr`, if `p_object` refers to an invalid object.
``` c++
Node3D *sp = Object::cast_to<Node3D>(ObjectDB::get_instance(p_object));
```